src/ext/ext.ts

```diff
@@ -1,28 +1,191 @@
+        show(properties: ContainerProperties) {
+          return properties.children.length <= 1;
+        },
+      },
+      {
+        translation: 'Common.Charts',
+        component: 'text',
+        show(properties: ContainerProperties) {
+          return properties.children.length > 1;
+        },
+      },
+      {
+        type: 'array',
+        ref: 'children',
+        itemTitleRef(properties: ContainerProperties) {
+          return properties.label;
+        },
+        allowAdd: true,
+        customAdd: true,
+        allowMove: true,
+        allowRemove: true,
+        addTranslation: 'Object.Container.Props.Add',
+        grouped: true,
+        remove(item: PropertiesChild, _properties: ContainerProperties, handler: PropertyHandler, args: PropertyArgs) {
+          handler.layout.qChildList.qItems.forEach((child: QChild) => {
+            if (
+              child.qData.containerChildId === item.refId ||
+              (child.qData.qExtendsId && child.qData.qExtendsId === item.refId)
+            ) {
+              containerHandler.removeChild(args.model, child.qInfo.qId);
+            }
+          });
+        },
+        add(_item: PropertiesChild, properties: ContainerProperties, handler: PropertyHandler, args: any) {
+          console.log('addChild child _item===', _item);
+          console.log('addChild child _data===', properties);
+          console.log('addChild child args===', args);
+          console.log('addChild child handler===', handler);
+          console.log('args.model.items==', args.model.items);
+          const button = document.querySelector('.pp-toplist-add-button');
+          let popoverWrapper = button?.querySelector('.add-chart-popover');
+          if (!popoverWrapper) {
+            popoverWrapper = document.createElement('div');
+            popoverWrapper.classList.add('add-chart-popover');
+            button?.appendChild(popoverWrapper);
+          }
+          containerHandler.addChild(args.model, popoverWrapper);
+        },
+        items: {
+          label: {
+            ref: 'label',
+            type: 'string',
+            translation: 'Object.Container.Props.Label',
+            expression: 'optional',
+            show: (item: PropertiesChild, handler: PropertyHandler) => {
+              return (
+                !item.isMaster ||
+                (item.isMaster &&
+                  containerHandler.isValidMaster(item.refId) &&
+                  !containerHandler.isAppPublished(handler.app))
+              );
+            },
+          },
+          condition: {
+            ref: 'condition',
+            translation: 'Object.Container.Props.ShowCondition',
+            type: 'string',
+            defaultValue: '',
+            expression: 'optional',
+            change(properties: ContainerProperties) {
+              // inject = if there isn't one
+              if (
+                properties.condition &&
+                !properties.condition.qStringExpression &&
```

Comment:
This conditions checks when the properties.condition exists but is not a qStringExpression.
With `!properties.condition?.qStringExpression`, it will return true when the `properties.condition` does not exist instead.